tests/installation/change_desktop.pm

```diff
@@ -43,7 +43,10 @@ sub change_desktop {
         }
         send_key 'ret';
     }
-    send_key_until_needlematch 'patterns-list-selected', 'tab', 10, 2;
+    # Give it some time to process the 'ret'
+    if (!check_screen('patterns-list-selected', 10)) {
+        send_key_until_needlematch 'patterns-list-selected', 'tab', 10, 2;
+    }
```

Comment:
If this is really needed then it sounds like going back to the old behaviour of send_key_until_needlematch is better. Can you think of a better way avoiding `check_screen` with non-zero timeout? You are saying "fix races" but I think you know just as well that introducing such time-dependant checks is *introducing* race conditions

Reply:
Well, `send_key_until_needlematch` itself is racy. If the key processing takes longer than the given `check_screen` timeout, it wouldn't work either.

I can think of a better way, but that's only something for next week. Instead of the `check_screen`, adding another `assert_screen 'pattern_selector';`, but that requires adding various new needles.

Reply:
Sure, it was racy in before. Just saying that you are not really "fixing" the race condition, it's just shifted. The assert_screen way sounds good. Would be great if you could to this, doesn't have to be rushed. New explicit needles are often better and even saves time because checking an additional needle takes like 0.5s when `check_screen` needs to wait the full 10s.

Reply:
> Sure, it was racy in before. Just saying that you are not really "fixing" the race condition, it's just shifted.

Right, the PR title is a bit misleading, but at least the commit messages should be accurate.

> The assert_screen way sounds good. Would be great if you could to this, doesn't have to be rushed. New explicit needles are often better and even saves time because checking an additional needle takes like 0.5s when check_screen needs to wait the full 10s.

Yep. I'll make a draft PR for that.

Reply:
I had a better idea: https://github.com/os-autoinst/os-autoinst-distri-opensuse/pull/15362